app/controllers/zip_exports_controller.rb

```diff
@@ -0,0 +1,28 @@
+class ZipExportsController < ApplicationController
+  before_action :load_var, only: :download
+  before_action :check_edit_permissions, only: :download
+
+  def download
+    if @zip_export.stored_on_s3?
+      redirect_to @zip_export.presigned_url(download: true), status: 307
+    else
+      send_file @zip_export.zip_file.path,
+                filename: URI.unescape(@zip_export.zip_file_file_name),
+                type: @zip_export.file_content_type
+    end
+  end
+
+  def file_expired
+  end
+
+  private
+
+  def load_var
+    @zip_export = ZipExport.find_by_id(params[:id])
+    redirect_to(file_expired_url, status: 301) and return unless @zip_export
+  end
+
+  def check_edit_permissions
+    render_403 unless @zip_export.user == current_user
```

Comment:
I know it's very basic, but this is how we did all other permissions. Check some of them out and you'll see, how basic they are, but we still put them in the Permissions helper class.